module/zfs/zfs_ioctl.c

```diff
@@ -5843,6 +5903,9 @@ zfs_ioctl_init(void)
 	zfs_ioctl_register_dataset_nolog(ZFS_IOC_USERSPACE_UPGRADE,
 	    zfs_ioc_userspace_upgrade, zfs_secpolicy_userspace_upgrade,
 	    POOL_CHECK_SUSPENDED | POOL_CHECK_READONLY);
+	zfs_ioctl_register_dataset_nolog(ZFS_IOC_USEROBJSPACE_UPGRADE,
+	    zfs_ioc_userobjspace_upgrade, zfs_secpolicy_userspace_upgrade,
+	    POOL_CHECK_SUSPENDED | POOL_CHECK_READONLY);
```

Comment:
Sure, as long as there's some mechanism in this patch to manually trigger the upgrade.  Needing to do this manually isn't something we expect/want most users to do.  Still since it may be needed I'd suggest adding this little hint to `zpool-features.5`.  Maybe a something like:

```
"This feature becomes \fBactive\fR as soon as it is enabled and will never return
to being \fBenabled\fR.  Each filesystem will be upgraded automatically when
mounted. The upgrade can be started manually on unmounted filesystems by running
`zfs set version=5 <pool/fs>`.  The upgrade process runs in the background and may
take a while to complete for filesystems containing a large number of files."
```

I'd expect the common case to be for users to enable this feature on a pool with filesystems already mounted.  In addition to the `dmu_dataset_own()` trigger let's add one to `dmu_objset_sync()` to get mounted filesystems.

```
if (dmu_objset_userobjspace_upgradable(os))
        dmu_objset_userobjspace_upgrade(*osp);
```